lib/value/src/kind/collection/index.rs

```diff
@@ -41,6 +70,26 @@ impl Collection<Index> {
             Some(self.min_length())
         }
     }
+
+    /// Removes the known value at the given index and shifts the
+    /// elements to the left
+    pub fn remove_shift(&mut self, index: usize) {
+        let min_length = self.min_length();
+        self.known_mut().remove(&index.into());
+        for i in index..min_length {
```

Comment:
What happens here if the collection has a gap in the indices, like `0`, `1`, `3`, `4`, and `1` is removed. Do we intentionally not shift the indices after the gap?